src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1132,10 +1133,81 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
     private boolean promptUser(PromptBehavior prompt) {
         return prompt == PromptBehavior.Always || prompt == PromptBehavior.REFRESH_SESSION;
     }
+    
+    /**
+     * check the redirectUri before sending the request
+     * if the redirectUri from the client does not match the valid redirectUri
+     * the client app would not jump to the login page
+     * redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE
+     * 
+     * @param callbackHandle
+     * @param request
+     * @throws UnsupportedEncodingException
+     * @return true if the RedictUri is valid
+     */
+    private boolean verifyBrokerRedirectUri(CallbackHandler callbackHandle, final AuthenticationRequest request){   
+        final String methodName = ":verifyBrokerRedirectUri";
+        String errMsg = new String();        
```

Comment:
Does it really make sense to make use the constructor here? Can't you just do String errMsg = "";